src/schema/course/utils.ts

```diff
@@ -35,3 +38,30 @@ export const buildCourseQuery = (params: QueryCourseListArgs) => {
     { key: 'audience_max_age_gt', value: params.audienceMaxAgeGt },
   ]);
 };
+
+export const getCoursesList = async ({
+  dataSources,
+  include,
+  ids,
+}: {
+  dataSources: any;
+  include: string[];
+  ids: string[];
+}) => {
+  const courses = await Promise.all(
+    ids.map(async (id) => {
+      try {
+        const query = buildCourseQuery({ include });
+        const course = await dataSources.courseAPI.getCourseDetails(id, query);
+        return normalizeKeys(course);
+      } catch (e) {
+        Sentry.captureException(e);
+        // eslint-disable-next-line no-console
+        console.error('error', e);
+        return null;
+      }
+    })
+  );
+
+  return courses.filter(Boolean);
```

Comment:
Are some values empty?

Reply:
The promises that return null are null so those need to be filtered out